### DISCUSSION TITLE
Why is Node Letter added to new Document Ids

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [ravendb/ravendb#16796] [Q&A]
We are working on upgrading our database and APIs from 3.x to 5.x and while testing today, I noticed that for new documents created the Node Letter ('A') was added to the end of the Document Ids. Why is this done? Is there a way to prevent this or configure this behavior away?

Instead of 'tickets-11043' it was created as 'tickets-11043-A'.  If this is the way the cloud-based Raven DB Servers work, we'll need to modify code that expects differently.

**Comment by DejanMilicic** [comment]
This behavior is intended to reduce the number of possible conflicts when creating new documents.
As is the case with other RavenDb features, this one is configurable.

Here is an example configuration that will prevent Node tag from being added to newly created ID
https://github.com/DejanMilicic/RavenWorkshop/blob/main/Northwind/Northwind/Features/Identifiers/CustomHiLoGenerator.cs#L70

**Comment by georgiosd** [comment]
The real question is: why do you care?

**Reply by mfonnesbeck** [reply]
We are using the number part of the identifier in external APIs as a code value to identify to 3rd parties. It was something developed long ago and the adding of it to the identifier messes how our application works.

**Comment by DejanMilicic** [comment]
Can you please explain in what sense your code "expects different" shapes of Identifiers? Are you parsing them in code?

**Reply by mfonnesbeck** [reply]
Since this is an identifier, it was assumed this would not change and previous developers created code that parsed the identifier, extracting out the numerical part of the identifier, and used it to create a code value that is used in sending to external partners and entities to identify an entity.

**Comment by mfonnesbeck** [comment]
Since this is an identifier, it was assumed this would not change and previous developers created code that parsed the identifier, extracting out the numerical part of the identifier, and used it to create a code value that is used in connection and identification with external partners.

**Reply by ayende** [reply]
You can convert the node tag into a numeric value and append it to the id, that will give you a numeric value
